hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/Constants.java

```diff
@@ -57,36 +49,27 @@
+            .put(ExecuteOperationResponse.class.getName(), ForwardCommand.class)
+            .put(DeployApplicationResponse.class.getName(), ForwardCommand.class)
+            .put(AddJdbcDriverResponse.class.getName(), ForwardCommand.class)
+            .put(AddDatasourceResponse.class.getName(), ForwardCommand.class)
+            .put(ExportJdrResponse.class.getName(), ForwardCommand.class)
+            .put(GenericErrorResponse.class.getName(), ForwardCommand.class);
 
     /**
      * These are the only valid commands that can come from UI clients.
      */
     ValidCommandsMap VALID_COMMANDS_FROM_UI = new ValidCommandsMap()
             .put(EchoCommand.REQUEST_CLASS.getName(), EchoCommand.class)
-            .put(DeployApplicationCommand.REQUEST_CLASS.getName(), DeployApplicationCommand.class)
-            .put(AddJdbcDriverCommand.REQUEST_CLASS.getName(), AddJdbcDriverCommand.class)
-            .put(AddDatasourceCommand.REQUEST_CLASS.getName(), AddDatasourceCommand.class)
-            .put(ExecuteOperationCommand.REQUEST_CLASS.getName(), ExecuteOperationCommand.class)
-            .put(ExportJdrCommand.REQUEST_CLASS.getName(), ExportJdrCommand.class)
-            .put(GenericErrorResponseCommand.REQUEST_CLASS.getName(), GenericErrorResponseCommand.class);
+            .put(DeployApplicationRequest.class.getName(), ForwardCommand.class)
+            .put(AddJdbcDriverRequest.class.getName(), ForwardCommand.class)
+            .put(AddDatasourceRequest.class.getName(), ForwardCommand.class)
+            .put(ExecuteOperationRequest.class.getName(), ForwardCommand.class)
+            .put(ExportJdrRequest.class.getName(), ForwardCommand.class)
+            .put(GenericErrorResponse.class.getName(), ForwardCommand.class);
 
     // QUEUES AND TOPICS
-    Endpoint DEST_FEED_EXECUTE_OP = new Endpoint(Type.QUEUE, "FeedExecuteOperation");
-    Endpoint DEST_FEED_DEPLOY_APPLICATION = new Endpoint(Type.QUEUE, "FeedDeployApplication");
-    Endpoint DEST_FEED_ADD_JDBC_DRIVER = new Endpoint(Type.QUEUE, "FeedAddJdbcDriver");
-    Endpoint DEST_FEED_ADD_DATASOURCE = new Endpoint(Type.QUEUE, "FeedAddDatasource");
-    Endpoint DEST_FEED_EXPORT_JDR = new Endpoint(Type.QUEUE, "ExportJDR");
-
-    Endpoint DEST_UICLIENT_EXECUTE_OP_RESPONSE = new Endpoint(Type.QUEUE, "UIClientExecuteOperationResponse");
-    Endpoint DEST_UICLIENT_DEPLOY_APPLICATION_RESPONSE = new Endpoint(Type.QUEUE, "UIDeployApplicationResponse");
-    Endpoint DEST_UICLIENT_ADD_JDBC_DRIVER_RESPONSE = new Endpoint(Type.QUEUE, "UIAddJdbcDriverResponse");
-    Endpoint DEST_UICLIENT_ADD_DATASOURCE_RESPONSE = new Endpoint(Type.QUEUE, "UIAddDatasourceResponse");
-    Endpoint DEST_UICLIENT_EXPORT_JDR_RESPONSE = new Endpoint(Type.QUEUE, "UIExportJDRResponse");
+    Endpoint FEED_COMMAND_QUEUE = new Endpoint(Type.QUEUE, "FeedCommandQueue");
+    Endpoint UI_COMMAND_QUEUE = new Endpoint(Type.QUEUE, "UiCommandQueue");
```

Comment:
`ForwardCommand` is gone in 4db0e03. There are a few commands in `BusCommands` and `WsCommands` there that handle the routing scenarios we have now, and new commands for new routing scenarios can be added easily.